github/util_v4_repository.go

```diff
@@ -41,11 +40,6 @@ func getRepositoryID(name string, meta interface{}) (githubv4.ID, error) {
 }
 
 func repositoryNodeIDExists(name string, meta interface{}) (bool, error) {
-	// Check if the name is a base 64 encoded node ID
-	_, err := base64.StdEncoding.DecodeString(name)
-	if err != nil {
-		return false, nil
-	}
```

Comment:
Are there any backwards compatibility risks with removing this? Aiming to read more on this front, but hesitating to merge this without more clarity around if this will be a breaking change for some.

Reply:
@jcudit That's what I wanted to verify with mocking out the Github API in the unit tests I wrote. I saw in the edit history that every version that accepted a node_id had this local check before calling out to the github api. Originally, it was checking the length. It was checking for base64, but I never saw a version without a pre-check, so it wasn't clear if it was originally put in to address an issue, or if the idea was to simply save an api call.

If you can think of any other cases that'd be worth testing for, I'd be more than happy to add them.

Reply:
> originally put in to address an issue

Yes, the aim was to allow passing in a repository name to improve the configuration experience.  See https://github.com/integrations/terraform-provider-github/pull/593 for details on this.

Reply:
c5a7393 suggests we check for both new and old formats.  How does everyone feel about this approach?  Seems like a 👎🏾 from a performance / rate limiting perspective but 👍🏾  for reliability.